### PR TITLE
changed esc functionality to remove last chronological filter, small clean

### PR DESCRIPTION
**Pull request by rexalex**
corresponds to issue #401.
- keep track of all filters applied or removed, chronologically in a new list
- if a previous same type filter is removed, then it will be removed from the list
- if a previous same type filter is changed, then the old one will be removed, the new one will be added as the most recent

### CODE REVIEW

File: assets/js/components/GlobalFilters.vue [hunk 1]
```diff
@@ -183,15 +179,35 @@ export default Vue.extend({
...
         }
       }
+
       return filters;
     },
   },
 
+  created() {
+    const self = this;
+    this.chronologicalFilters = [];
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: i would call this `this.filter_stack` or filters_stack or meh, filterStack

File: assets/js/components/GlobalFilters.vue [hunk 2]
```diff
@@ -183,15 +179,35 @@ export default Vue.extend({
...
+
+    this.makeChronologicalFiltersFromExisting();
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: and this one `this.initFilterStack()`

File: assets/js/components/GlobalFilters.vue [hunk 3]
```diff
@@ -183,15 +179,35 @@ export default Vue.extend({
...
+    this.handleEsc();
+  },
 
   methods: {
-    removeFilter(e){
-      const remove_el = e.target.getAttribute('filter')
-      this.filters[remove_el] = null;
+    makeChronologicalFiltersFromExisting() {
+      for(var i in this.filters) {
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: ES6 please,
```js
for(const i in ...)
```
Actually, for const `k` from key, `i` makes one expect a numeric index.

File: assets/js/components/GlobalFilters.vue [hunk 4]
```diff
@@ -183,15 +179,35 @@ export default Vue.extend({
...
+        if(this.filters[i]) {
+          this.chronologicalFilters.push({type: i, val: this.filters[i]});
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: why do you care about the value? did you want to build a full undo system?

File: assets/js/components/GlobalFilters.vue [hunk 5]
```diff
@@ -206,29 +222,47 @@ export default Vue.extend({
...
-      window.addEventListener("keyup", function(e){
-        if (e.keyCode == 27) {
-          if($this.$el.querySelector('.list-filters')){
-          let last_filter = $this.$el.querySelector('.list-filters .filter-item:last-child');
-          if (!last_filter) return;
-          this.filters[last_filter.getAttribute('filter')] = null
-          }
-        }
-      })
+      let removedFilter = this.chronologicalFilters[this.chronologicalFilters.length-1];
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: so this isn't a full undo.
why not simply `.pop()`? the last filter still remains in the stack, this doesn't look right.

File: assets/js/components/GlobalFilters.vue [hunk 6]
```diff
@@ -206,29 +222,47 @@ export default Vue.extend({
...
       }
+      this.chronologicalFilters = [];
+    },
+    // it is executed each time this.filters changes
+    // it will keep track of all filters applied or removed, chronologically
+    // if a previous same type filter is removed, then it will be removed from the list
+    // but if it is changed after remove, the new one will be added as the most recent
+    handleFilter(type, val, old) {
+      if(old) {
+        this.removeOneFilter(type, val);
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: you don't need to pass `val`

File: assets/js/components/GlobalFilters.vue [hunk 7]
```diff
@@ -206,29 +222,47 @@ export default Vue.extend({
...
+          this.chronologicalFilters.push({type, val});
+        }
+      }
     },
-  },
 
-  handleFilter(type, val, old) {
-    console.log(type, val, old)
+    removeOneFilter(type, val) {
+      for(var i=0; i<this.chronologicalFilters.length; i++) {
```
Comment (xlotlu) [inline assets/js/components/GlobalFilters.vue]: if you loop, at least break on first find.
but there's no need to loop: either make this a simple stack and just `.indexOf()`, or else use `array.find()`. it's new in ES6 and we have it polyfilled.

File: assets/js/components/Sectors.vue [hunk 8]
```diff
@@ -810,9 +810,9 @@ export default Chart.extend({
         // we can't use this._arc directly as it yields funky distortions,
         // so we need a custom interpolation. attrTween to the rescue
         .attrTween('d', function(d) {
-	  const interpolate = d3.interpolate(
```
Comment (xlotlu) [inline assets/js/components/Sectors.vue]: the cleanup is nice, but it would be nicer not to mix it with unrelated commits

File: assets/js/components/GlobalFilters.vue (hunk 5 above)
Comment (rexalex) [reply assets/js/components/GlobalFilters.vue]: The pop() method removes the last element from an array and returns that element. This method changes the length of the array.
Changes to this.filters call handleFilter, if chronologicalFilters are also handled elsewhere, then in handleFilter they be handled again.
Any changes necessary to chronologicalFilters should be done only in handleFilter.

File: assets/js/components/GlobalFilters.vue (hunk 6 above)
Comment (rexalex) [reply assets/js/components/GlobalFilters.vue]: will remove val

File: assets/js/components/GlobalFilters.vue (hunk 4 above)
Comment (rexalex) [reply assets/js/components/GlobalFilters.vue]: you are right, they are not needed, used them only for debugging. will remove

File: assets/js/components/GlobalFilters.vue (hunk 5 above)
Comment (xlotlu) [reply assets/js/components/GlobalFilters.vue]: right
